### PR TITLE
:window: :wrench: Use PATCH Api for toggling connections

### PR DESCRIPTION
**Pull request by timroes**
## What

Part of https://github.com/airbytehq/airbyte/issues/15669

To prepare for removing the `syncCatalog` from the connection list API, we're switching over the enable/disable switch in the table to no longer send the full connection object as an update, but only the `status`.

This also introduced proper loading state for the switch inside the table.

## How

This creates a new `useEnableConnection` hook, which can be used in addition to the `useUpdateConnection` to just enable/disable the connection, which will also trigger the correct analytics event in this case.

This moves the logic to toggle down into the StatusCell, to no longer need to property drill through several layers there (which was originally done to have access to the `connection`, which we won't need now anymore with the update being PATCH and the analytic event moved to the hook). We removed two properties from the event as well, since they require the `syncCatalog` which is planned to get rid of in this place (see originally linked issue).

### CODE REVIEW

File: airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx [hunk 1]
```diff
@@ -23,16 +24,21 @@ const ProgressMessage = styled.div`
   padding: 7px 0;
 `;
 
-const StatusCell: React.FC<IProps> = ({ enabled, isManual, id, onChangeStatus, isSyncing, onSync, allowSync }) => {
+const StatusCell: React.FC<IProps> = ({ enabled, isManual, id, isSyncing, onSync, allowSync }) => {
+  const { mutateAsync: enableConnection, isLoading } = useEnableConnection();
+
   const [{ loading }, OnLaunch] = useAsyncFn(async (event: React.SyntheticEvent) => {
     event.stopPropagation();
-    await onSync(id);
```
Comment (timroes) [inline airbyte-webapp/src/components/EntityTable/components/StatusCell.tsx:29]: :information_source: This method wasn't `async`.

File: airbyte-webapp/src/hooks/services/useConnectionHook.tsx [hunk 2]
```diff
@@ -201,10 +202,49 @@ const useUpdateConnection = () => {
   return useMutation((connectionUpdate: WebBackendConnectionUpdate) => service.update(connectionUpdate), {
     onSuccess: (connection) => {
       queryClient.setQueryData(connectionsKeys.detail(connection.connectionId), connection);
+      // Update the connection inside the connections list response
+      queryClient.setQueryData<ListConnection>(connectionsKeys.lists(), (ls) => ({
+        ...ls,
+        connections:
+          ls?.connections.map((conn) => {
+            if (conn.connectionId === connection.connectionId) {
```
Comment (lmossman) [inline airbyte-webapp/src/hooks/services/useConnectionHook.tsx:210]: This code might be slightly more readable if `connection` was called something like `updatedConnection` instead

File: airbyte-webapp/src/hooks/services/useConnectionHook.tsx [hunk 3]
```diff
@@ -226,10 +266,6 @@ const useConnectionList = (): ListConnection => {
   return useSuspenseQuery(connectionsKeys.lists(), () => service.list(workspace.workspaceId));
 };
 
-const invalidateConnectionsList = async (queryClient: QueryClient) => {
-  await queryClient.invalidateQueries(connectionsKeys.lists());
-};
```
[the remark this answers is not in the document]
Comment (lmossman) [reply airbyte-webapp/src/hooks/services/useConnectionHook.tsx:231]: The new changes seem like a much cleaner and more efficient way to do it!

**Review by lmossman — APPROVED** [review]
LGTM. Tested locally and it seems to be working as expected

**Pull request merged**